runtime/tchannel_server.go

```diff
@@ -141,13 +141,20 @@ func NewTChannelRouter(registrar tchannel.Registrar, g *Gateway) *TChannelRouter
 }
 
 // Register registers the given TChannelEndpoint.
-func (s *TChannelRouter) Register(e *TChannelEndpoint) {
+func (s *TChannelRouter) Register(e *TChannelEndpoint) error {
```

Comment:
Yes, `Register` is done in one goroutine, the locks in tchannel register is over kill for our use case, but since it's initialization only, so not a big deal.